src/libraries/PositionConfig.sol

```diff
@@ -3,17 +3,61 @@ pragma solidity ^0.8.24;
 
 import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
 
-// A PositionConfig is the input for creating and modifying a Position in core, set per tokenId
+// A PositionConfig is the input for creating and modifying a Position in core, whos truncated hash is set per tokenId
 struct PositionConfig {
     PoolKey poolKey;
     int24 tickLower;
     int24 tickUpper;
 }
 
-/// @notice Library for computing the configId given a PositionConfig
+/// @notice Library to get and set the PositionConfigId and subscriber status for a given tokenId
 library PositionConfigLibrary {
+    using PositionConfigLibrary for PositionConfig;
+
+    bytes32 constant MASK_UPPER_BIT = 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
+    bytes32 constant DIRTY_UPPER_BIT = 0x8000000000000000000000000000000000000000000000000000000000000000;
+
+    /// @notice returns the truncated hash of the PositionConfig for a given tokenId
+    function getConfigId(mapping(uint256 => bytes32) storage positionConfigs, uint256 tokenId)
+        internal
+        view
+        returns (bytes32 configId)
+    {
+        bytes32 _config = positionConfigs[tokenId];
+        configId = _config & MASK_UPPER_BIT;
+    }
+
+    function setConfigId(
+        mapping(uint256 => bytes32) storage positionConfigs,
+        uint256 tokenId,
+        PositionConfig calldata config
+    ) internal {
+        positionConfigs[tokenId] = config.toId();
+    }
+
+    function setSubscribe(mapping(uint256 => bytes32) storage positionConfigs, uint256 tokenId) internal {
+        bytes32 _config = positionConfigs[tokenId];
+        positionConfigs[tokenId] = _config | DIRTY_UPPER_BIT;
+    }
+
+    function setUnsubscribe(mapping(uint256 => bytes32) storage positionConfigs, uint256 tokenId) internal {
+        bytes32 _config = positionConfigs[tokenId];
+        positionConfigs[tokenId] = _config & MASK_UPPER_BIT;
```

Comment:
probably cheaper not to have an intermediate var?
`positionConfigs[tokenId] &= MASK_UPPER_BIT`
and above
`positionConfigs[tokenId] |= DIRTY_UPPER_BIT;`

Reply:
weeeeeird its in [the docs](https://docs.soliditylang.org/en/latest/types.html#compound-and-increment-decrement-operators)

ok so either way i'd say get rid of `_config` from memory and just do 
```
positionConfigs[tokenId] = positionConfigs[tokenId] & MASK_UPPER_BIT;
```

Reply:
wait I'm sorry - it does! I must have been doing something differently yesterday 💤  i was tired.. it was 1030 lol